### ISSUE TITLE
Samples from associated projects cannot be added to the cart

### ISSUE THREAD

**Issue by apetkau** [phac-nml/irida#1392]
## Describe the bug

Samples from associated projects are unable to be properly added to the cart.  You use the option **Add to Cart** and they will appear to go into the cart.

![image](https://user-images.githubusercontent.com/200517/198710653-823984b4-3e89-41cc-b32a-e33e88e68ed5.png)

But after clicking on the cart you will be presented with a message _Woops..Nothing in here!_.

![image](https://user-images.githubusercontent.com/200517/198710420-17cb364c-cc3e-43fb-84df-24a93980a808.png)

This results in the below exception:

```
2022-10-28 13:51:32.151 ERROR 20873 --- [nio-8080-exec-4] .c.b.i.r.w.e.IridaCustomExceptionHandler : Join between the project and this identifier doesn't exist

ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException: Join between the project and this identifier doesn't exist
        at ca.corefacility.bioinformatics.irida.service.impl.sample.SampleServiceImpl.getSampleForProject(SampleServiceImpl.java:303) ~[classes/:na]
        at ca.corefacility.bioinformatics.irida.service.impl.sample.SampleServiceImpl$$FastClassBySpringCGLIB$$9792ab2e.invoke(<generated>) ~[classes/:na]
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.23.jar:5.3.23]
...
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat8-coyote-8.5.39.jar:8.5.39]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat8-util-8.5.39.jar:8.5.39]
        at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
```

You also cannot delete the samples in the cart or clear the cart anymore. You also cannot re-add samples individually from each project.

## Steps to reproduce the problem
What were you doing when you encountered the problem?
1. Create two projects with a single sample.
2. Create an associated project that includes the above two projects.
3. Select samples from this project with associated projects.
4. Add samples to cart.
5. Click on cart. You will see that the cart says it's empty.

## Expected behaviour

I expected the samples to be displayed within the cart when clicking the cart.

## Additional context

Happens both with IRIDA `22.05.5` and `22.09.1` (latest release as of this issue).

A related issue appears to be https://github.com/phac-nml/irida/issues/331, but this seems to have been resolved in 2019.

**Comment by ericenns** [comment]
I found a way of removing the samples from the cart, but it is certainly a little weird and leads to some strange behaviour.

Steps:

1) Go to project 5 (id 4) in the hibernate seeded data, select project filter to show project 2 and then select the first sample in project 2 and click add the cart.
![image](https://user-images.githubusercontent.com/492127/198714018-a1994d95-e45c-45c3-a7e8-cc37967a68bf.png)

2) Sample shows added to cart
![image](https://user-images.githubusercontent.com/492127/198714106-e7c1cdd0-dd41-4ab4-b2eb-9064aef141e0.png)

3) Going to cart page shows no samples
![image](https://user-images.githubusercontent.com/492127/198714173-bb85b4bc-10e5-4966-932a-3846d6f73788.png)

4) Go back to project 5 (id 4), select project filter to show project 2 and then click on the name of the first sample in project 2. This launches the sample modal but shows the wrong project. Then click remove sample.
![image](https://user-images.githubusercontent.com/492127/198714241-4deb04c0-dd8e-4375-b775-51864c32c558.png)

5) Sample is successfully removed from the cart.
![image](https://user-images.githubusercontent.com/492127/198714292-6488f5c3-806c-4dd6-be02-9d9036c6d9cd.png)

6) After dismissing the sample modal on re-opening it still shows the remove from cart button and the wrong project (project 5 instead of project 2).
![image](https://user-images.githubusercontent.com/492127/198714356-149dade1-3672-4180-8ffc-565cb2b0d615.png)

7) Refreshing the page and re-applying the filter and then clicking on the sample launches the modal with the correct project.
![image](https://user-images.githubusercontent.com/492127/198714420-d788400c-ae2d-4207-b87f-3ac2ffaeb9dd.png)

**Comment by ericenns** [comment]
Fixed in #1395
